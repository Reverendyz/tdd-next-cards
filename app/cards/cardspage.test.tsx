import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import CardsPage from '@/app/cards/page';

describe('CardsPage', () => {
  test('renders no cards initially', () => {
    const { asFragment } = render(<CardsPage />);
    
    expect(asFragment()).toMatchSnapshot();
    
    const cards = screen.queryAllByRole('heading', { level: 2 });
    expect(cards.length).toBe(0);
  });

  test('adds a new card via form submission and matches snapshot', () => {
    const { asFragment } = render(<CardsPage />);
    
    const titleInput = screen.getByPlaceholderText(/title/i) as HTMLInputElement;
    const descriptionInput = screen.getByPlaceholderText(/description/i) as HTMLInputElement;
    const addButton = screen.getByRole('button', { name: /add card/i });
    
    fireEvent.change(titleInput, { target: { value: 'New Title' } });
    fireEvent.change(descriptionInput, { target: { value: 'New Description' } });
    fireEvent.click(addButton);

    // Snapshot after adding a card
    expect(asFragment()).toMatchSnapshot();
    
    const cards = screen.getAllByRole('heading', { level: 2 });
    expect(cards.length).toBe(1);
    expect(cards[0]).toHaveTextContent('New Title');
  });

  test('does not add a card if form is incomplete', () => {
    render(<CardsPage />);
    
    const titleInput = screen.getByPlaceholderText(/title/i) as HTMLInputElement;
    const addButton = screen.getByRole('button', { name: /add card/i });
    
    fireEvent.change(titleInput, { target: { value: 'Only Title' } });
    fireEvent.click(addButton);
    
    const cards = screen.queryAllByRole('heading', { level: 2 });
    expect(cards.length).toBe(0);
  });

  test('form clears after successful submission', () => {
    render(<CardsPage />);
    
    const titleInput = screen.getByPlaceholderText(/title/i) as HTMLInputElement;
    const descriptionInput = screen.getByPlaceholderText(/description/i) as HTMLInputElement;
    const addButton = screen.getByRole('button', { name: /add card/i });
    
    fireEvent.change(titleInput, { target: { value: 'Another Title' } });
    fireEvent.change(descriptionInput, { target: { value: 'Another Description' } });
    fireEvent.click(addButton);
    
    expect(titleInput.value).toBe('');
    expect(descriptionInput.value).toBe('');
  });
});
