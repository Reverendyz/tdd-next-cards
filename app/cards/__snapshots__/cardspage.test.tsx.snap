// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsPage adds a new card via form submission and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20"
  >
    <div
      class="text-center mb-8"
    >
      <h1
        class="text-2xl font-bold"
      >
        Cards
      </h1>
      <p
        class="text-gray-500"
      >
        Dynamic cards with random content
      </p>
    </div>
    <form
      class="mb-8 text-center"
    >
      <input
        class="border p-2 rounded mr-2 text-black font-bold"
        name="title"
        placeholder="Title"
        type="text"
        value=""
      />
      <input
        class="border p-2 rounded mr-2 text-black font-bold\`"
        name="description"
        placeholder="Description"
        type="text"
        value=""
      />
      <button
        class="px-6 py-2 bg-blue-500 text-white rounded hover:bg-blue-600"
        type="submit"
      >
        Add Card
      </button>
    </form>
    <div
      class="grid grid-cols-1 sm:grid-cols-3 gap-8"
    >
      <div
        class="p-4 border rounded shadow-md"
      >
        <h2
          class="text-xl font-semibold"
        >
          New Title
        </h2>
        <p
          class="text-gray-600"
        >
          New Description
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardsPage renders no cards initially 1`] = `
<DocumentFragment>
  <div
    class="items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20"
  >
    <div
      class="text-center mb-8"
    >
      <h1
        class="text-2xl font-bold"
      >
        Cards
      </h1>
      <p
        class="text-gray-500"
      >
        Dynamic cards with random content
      </p>
    </div>
    <form
      class="mb-8 text-center"
    >
      <input
        class="border p-2 rounded mr-2 text-black font-bold"
        name="title"
        placeholder="Title"
        type="text"
        value=""
      />
      <input
        class="border p-2 rounded mr-2 text-black font-bold\`"
        name="description"
        placeholder="Description"
        type="text"
        value=""
      />
      <button
        class="px-6 py-2 bg-blue-500 text-white rounded hover:bg-blue-600"
        type="submit"
      >
        Add Card
      </button>
    </form>
    <div
      class="grid grid-cols-1 sm:grid-cols-3 gap-8"
    />
  </div>
</DocumentFragment>
`;
